src/main/kotlin/io/snyk/plugin/services/SnykTaskQueueService.kt

```diff
@@ -74,16 +81,69 @@ class SnykTaskQueueService(val project: Project) {
 
                 if (settings.ossScanEnable) {
                     if (!getOssService(project).isCliInstalled()) downloadLatestRelease()
+                    waitForCliDownload()
                     scheduleOssScan()
                 }
                 if (settings.snykCodeSecurityIssuesScanEnable || settings.snykCodeQualityIssuesScanEnable) {
                     scheduleSnykCodeScan()
                 }
                 if (isIacEnabled() && settings.iacScanEnabled) {
-                    //fixme: downloadLatestRelease() should check if downloading is in process
                     if (!getIacService(project).isCliInstalled()) downloadLatestRelease()
+                    waitForCliDownload()
                     scheduleIacScan()
                 }
+
+                if (isContainerEnabled() && settings.containerScanEnabled) {
+                    if (!getContainerService(project).isCliInstalled()) downloadLatestRelease()
+                    waitForCliDownload()
+                    scheduleContainerScan()
+                }
+            }
+        })
+    }
+
+    private fun waitForCliDownload() {
+        if (!isCliDownloading()) return
+
+        getAppScheduledExecutorService().submit(
+            Callable {
+                while (isCliDownloading()) {
+                    Thread.sleep(waitForDownloadMillis)
+                }
+            }
+        ).get()
+    }
+
+    private fun scheduleContainerScan() {
+        taskQueueIac.run(object : Task.Backgroundable(project, "Snyk Container is scanning...", true) {
```

Comment:
we need taskQueueContainer
otherwise we'll pospone Container scan untill shedulled IaC scan finished